python/GafferImageUI/CatalogueUI.py

```diff
@@ -481,15 +494,18 @@ def __removeClicked( self, *unused ) :
 			return
 
 		with Gaffer.UndoScope( self.getPlug().ancestor( Gaffer.ScriptNode ) ) :
+
+			orderedImages = self.__orderedImages()
 			for index in reversed( sorted( indices ) ) :
-				metadataIndex = Gaffer.Metadata.value( self.__images()[index], _ImagesPath.indexMetadataName )
-				self.__images().removeChild( self.__images()[index] )
+				image = self.__images()[ index ]
+				logicalIndex = orderedImages.index( image )
```

Comment:
Again, this isn't what I understand by "logical index".

Reply:
https://github.com/GafferHQ/gaffer/commit/9aed9541e19f8838e8997a7eb8c62f5803530915